C-Chapter0/src/count.cpp

```diff
@@ -0,0 +1,25 @@
+/*
+OVERVIEW:  Given 2 integers (num1, num2), write a function that returns the count of numbers between the num1..num2 that are divisible by num1.
+		E.g.: count(3, 15) returns 5 (3, 6, 9, 12, 15).
+		Note : You need to consider Inclusive range
+INPUTS:  Two numbers num1, num2.
+		num1, num2 >= 0.
+
+OUTPUT: Return the count of numbers between the num1..num2 that are divisible by num1.
+
+ERROR CASES: Return -1 in error cases.
+
+NOTES:
+*/
+int count(int num1, int num2) {
+	if (num1 == 0)
+		return -1;
+	if (num2 == 0)
+		return 0;
+	if (num1 >= num2)
+		return 1;
+	if (num2%num1 == 0)
+		return 1 + count(num1, num2 - 1);
+
```

Comment:
Good to see that you're using recursion for this. Nice approach. Can you discuss any other approaches that you've thought of here in comments.

Reply:
An another approach I thought was of course a loop solution.

Reply:
Don't go by logic. Think if you can have any mathematical solution that can get you the answer by O(1)?